### ISSUE TITLE
Code Release

### ISSUE THREAD

**Issue by thom966** [jiaweihe1996/BA-Det#1]
Dear author, 

When will be the code released?

**Comment by SPA-junghokim** [comment]
Dear author, 

I wanted to express my sincere gratitude for your research work. Your contributions have been truly inspiring and have been instrumental in advancing the field.

In addition to your research, I am deeply interested in exploring the realm of monocular camera-based 3D object recognition technology for autonomous driving. I believe that the insights gained from your work could greatly benefit my own endeavors in this area.

I am curious to know if you have any plans to release the code or any related resources for your research project. If possible, could you kindly inform me of the expected timeline for the code release? This information would greatly aid me in planning and conducting my own research.

Thank you once again for your invaluable contributions to the field. I look forward to your response and hope for the possibility of collaborating in the future.

**Comment by jiaweihe1996** [comment]
Sorry for the delay in releasing the code. This is because the code for BA-Det is intricately connected to the subsequent extension work. We will try hard to finish the extension work and release the code this month. We will let you know as soon as we make it publicly available.

**Comment by SPA-junghokim** [comment]
Thank you for your prompt response. I greatly appreciate the information provided, and I'm eagerly looking forward to that day when more insights are unveiled.

I have one more question that arose while contemplating the previous discussion. Regarding the usage of oba loss during training, is it utilized solely as an auxiliary loss? It appears that during inference, feature matching takes place with oba. Could you kindly shed some light on the post-feature matching process during inference?

Your clarification on this matter would be highly valuable in deepening my understanding of the models.

Once again, thank you for your time and consideration. I am genuinely grateful for your willingness to share your expertise.

**Comment by jiaweihe1996** [comment]
> Thank you for your prompt response. I greatly appreciate the information provided, and I'm eagerly looking forward to that day when more insights are unveiled.
> 
> I have one more question that arose while contemplating the previous discussion. Regarding the usage of oba loss during training, is it utilized solely as an auxiliary loss? It appears that during inference, feature matching takes place with oba. Could you kindly shed some light on the post-feature matching process during inference?
> 
> Your clarification on this matter would be highly valuable in deepening my understanding of the models.
> 
> Once again, thank you for your time and consideration. I am genuinely grateful for your willingness to share your expertise.

OBA loss is not just an auxiliary loss.
The optimization objectiveness in training and inference both minimize reprojection error in bundle adjustment.  Intuitively, during the training stage, we suppose the object pose is given and optimize the correspondence matching. During the inference, given the feature correspondence, we optimize the object pose.

**Comment by SPA-junghokim** [comment]
Thank you for your reply.

I wonder about the inference. During inference, if it is given the feature correspondence, how to use the correspondence map? After given the feature correspondence map, is it used for refining 1-stage 3d box result??

**Comment by jiaweihe1996** [comment]
> Thank you for your reply.
> 
> I wonder about the inference. During inference, if it is given the feature correspondence, how to use the correspondence map? After given the feature correspondence map, is it used for refining 1-stage 3d box result??

Yes. correspondence is considered as the observation in calculating reprojection error. The optimization of object pose is initialized from 1-stage 3d box result.

**Comment by cz19860416** [comment]
> Sorry for the delay in releasing the code. This is because the code for BA-Det is intricately connected to the subsequent extension work. We will try hard to finish the extension work and release the code this month. We will let you know as soon as we make it publicly available.

Hi, Thank you for your great work, just wonder, do you still have plan to release the code?

**Comment by cz19860416** [comment]
> Sorry for the delay in releasing the code. This is because the code for BA-Det is intricately connected to the subsequent extension work. We will try hard to finish the extension work and release the code this month. We will let you know as soon as we make it publicly available.

Hello, did you still have the plan to release the code ? Thank you :)
